master/buildbot/steps/source/bzr.py

```diff
@@ -50,7 +50,9 @@ def __init__(self, repourl=None, baseURL=None, mode='incremental',
         if baseURL is not None and defaultBranch is None:
             raise ValueError("you must provide defaultBranch with baseURL")
 
-        assert self.mode in ['incremental', 'full']
+        if not self._hasAttrGroupMember('mode', self.mode):
```

Comment:
It sounds like this is a nice place to break off this PR and start on a new one.  I don't have any problem with work to make the source steps more consistent!